Makefile: make PROJ compile on Publish Bleeding Edge

otan opened the pull request:
As docgen relies on builtins which relies on proj, we are stuck in a
situation in cross compilation where we are trying to install docgen
with a non-native libproj.a file causing the linker to fail in `Upload
Binaries`. Fix this by not compiling docgen on cross compilations.

Also fix the windows compilation as PROJ will dump it into a
libproj_4_9.a instead of libproj.a.

Also make execgen not depend on LIBPROJ since it's not required.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49917)
<!-- Reviewable:end -->

otan commented:
bleeding edge: https://teamcity.cockroachdb.com/viewType.html?buildTypeId=Cockroach_MergeToMaster&branch_Cockroach=49917&tab=buildTypeStatusDiv

otan committed:
Makefile: make PROJ compile on Publish Bleeding Edge

As docgen relies on builtins which relies on proj, we are stuck in a
situation in cross compilation where we are trying to install docgen
with a non-native libproj.a file causing the linker to fail in `Upload
Binaries`. Fix this by not compiling docgen on cross compilations.

Also fix the windows compilation as PROJ will dump it into a
libproj_4_9.a instead of libproj.a. Cannot see an option to fix this
on the CMakeLists.txt.

Also make execgen not depend on LIBPROJ.

Release note: None

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

Did execgen never really require LIBPROJ? I'm not seeing what changed that allowed that.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49917#-:-M9K1FSs0DAVjXpKE7Jd:b-ua8tx7)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @petermattis)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=petermattis

> Did execgen never really require LIBPROJ? I'm not seeing what changed that allowed that.

https://github.com/cockroachdb/cockroach/pull/49783/files did 🙈 but i was focusing on docgen, not execgen. woops!
